docs_src/security/V3Migration.md

```diff
@@ -1,3 +1,118 @@
+the impact of the recent microservice authentication changes,
+which primarily affect EdgeX internals, is not reflected in the threat model.
+
+## API Gateway Breaking Authentication Changes
+
+In EdgeX 2.0, the `secrets-config` utility was used to create a user account in the API gateway (Kong)
+and associate it to a user-specified public key.
+A user would then self-create a JWT, and use it for authentication against the API gateway.
+These tokens were opaque to EdgeX microservices because their contents were controlled by the user,
+and only the API gateway had the information needed to validate them.
+
+In EdgeX 3.0, the `secrets-config` utility is still used to create a user account,
+but instead of creating it in the API gateway,
+the user account is created in the EdgeX secret store,
+and the Vault identity secrets engine is used to generate and verify JWT's.
+All EdgeX services implicitly trust the EdgeX secret store
+and have a secret store token issued to them at startup
+that can be used to request a JWT from Vault.
+
+Externally-originated requests are performed similarly to how they were done before:
+provide the JWT in the `Authorization` header
+and direct the request at the API gateway with a path prefix denoting the desired service.
+The key difference is in obtaining the JWT.
+In EdgeX 2.0, the client simply generated the JWT using its private key.
+In EdgeX 3.0, obtaining a JWT is a two-step process.
+First, authenticate to the EdgeX secret store (Vault) to obtain a secret store token.
+Second, exchange the secret store token for a JWT.
+This process is described in detail in the
+[authenticating chapter](https://docs.edgexfoundry.org/3.0/security/Ch-Authenticating/)
+of the EdgeX documentation.
+Due to these changes, the `secrets-config proxy jwt` helper command has been removed.
+This same chapter also explains that, similar to Kong,
+Vault has an extensible authentication mechanism,
+although only username/password (with a randomized strong password)
+is enabled out of the box.
+
+As was before, all requests (with the exception of a passthrough for Vault authentication)
+are checked at the API gateway prior to forwarding to the backend service for fulfillment.
+
+## Microservice-level Breaking Authentication Changes
+
+EdgeX microservices in EdgeX 3.0 will now require authentication on a per-route basis,
+even for requests that originate behind the API gateway.
+Peer-to-peer service requests (such as a device service calling core-metadata,
+or core-command forwarding a request to a device service) are authenticated automatically.
+This new behavior may create compatibility issues for custom components
+that worked fine in EdgeX 2.0 that may suddenly experience authentication failures in EdgeX 3.0.
+This new behavior may also create issues for 3rd party components,
+such as the eKuiper rules engine,
+because of its ability to issue ad-hoc HTTP requests in response to certain events.
```

Comment:
Add link to eKuiper section in main V3 migration guide here: https://docs.edgexfoundry.org/3.0/V3TopLevelMigration/#secure-mode

Reply:
Fixed.